cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1530,6 +1530,14 @@ if (vecgeom)
   endif()
 endif()
 
+#---Check for protobuf-------------------------------------------------------------------
+find_package(Protobuf)
```

Comment:
I would prefer to have `protobuf` checked as an optional dependency only if `tmva-sofie` option is ON, now you check protobuf package even if `tmva-sofie` is OFF. Another thing to add, it is a missing 'fail-on-missing' check. Please check for an inspiration MySQL example (in your case it will be `tmva-sofie` and Protobuf, where you can add as well easily a version check): https://github.com/root-project/root/blob/master/cmake/modules/SearchInstalledSoftware.cmake#L660-L672